VS2013 compiler has a lot of non-error warnings while compiling vscript.lib

In the VS2013 compiler, compiling `vscript.lib` causes several warnings stemming from the Squirrel API. It still compiles, but it's very likely to confuse SDK users who are used to warnings which cause compilation to fail entirely.

Here is a list of every warning given on my end, with irrelevant parts of the file paths cut out:
```
1>e:\...\sp\src\vscript\squirrel\squirrel\sqclass.cpp(192): warning C4100: 'v' : unreferenced formal parameter
1>e:\...\sp\src\vscript\squirrel\squirrel\sqcompiler.cpp(177): warning C4611: interaction between '_setjmp' and C++ object destruction is non-portable
1>e:\...\sp\src\vscript\squirrel\squirrel\sqapi.cpp(183): warning C4100: 'v' : unreferenced formal parameter
1>e:\...\sp\src\vscript\squirrel\squirrel\sqcompiler.cpp(895): warning C4127: conditional expression is constant
1>e:\...\sp\src\vscript\squirrel\squirrel\sqcompiler.cpp(1061): warning C4127: conditional expression is constant
1>e:\...\sp\src\vscript\squirrel\squirrel\sqcompiler.cpp(1071): warning C4127: conditional expression is constant
1>e:\...\sp\src\vscript\squirrel\squirrel\sqbaselib.cpp(44): warning C4100: 'v' : unreferenced formal parameter
1>e:\...\sp\src\vscript\squirrel\squirrel\sqbaselib.cpp(823): warning C4100: 'r' : unreferenced formal parameter
1>e:\...\sp\src\vscript\squirrel\squirrel\sqbaselib.cpp(823): warning C4100: 'l' : unreferenced formal parameter
1>e:\...\sp\src\vscript\squirrel\squirrel\sqbaselib.cpp(953): warning C4244: '=' : conversion from 'int' to 'SQChar', possible loss of data
1>e:\...\sp\src\vscript\squirrel\squirrel\sqbaselib.cpp(954): warning C4244: '=' : conversion from 'int' to 'SQChar', possible loss of data
1>e:\...\sp\src\vscript\squirrel\squirrel\sqapi.cpp(57): warning C4702: unreachable code
1>e:\...\sp\src\vscript\squirrel\squirrel\sqbaselib.cpp(136): warning C4706: assignment within conditional expression
1>e:\...\sp\src\vscript\squirrel\squirrel\sqmem.cpp(8): warning C4100: 'oldsize' : unreferenced formal parameter
1>e:\...\sp\src\vscript\squirrel\squirrel\sqmem.cpp(10): warning C4100: 'size' : unreferenced formal parameter
1>e:\...\sp\src\vscript\vscript_squirrel.cpp(805): warning C4800: 'SQBool' : forcing value to bool 'true' or 'false' (performance warning)
1>e:\...\sp\src\vscript\vscript_squirrel.cpp(914): warning C4800: 'SQBool' : forcing value to bool 'true' or 'false' (performance warning)
1>e:\...\sp\src\vscript\squirrel\squirrel\sqlexer.cpp(511): warning C4244: 'argument' : conversion from 'int' to 'const SQChar', possible loss of data
1>e:\...\sp\src\vscript\squirrel\squirrel\sqtable.h(87): warning C4706: assignment within conditional expression
1>e:\...\sp\src\vscript\squirrel\squirrel\sqlexer.cpp(276): warning C4702: unreachable code
1>e:\...\sp\src\vscript\squirrel\sqstdlib\sqstdaux.cpp(34): warning C4706: assignment within conditional expression
1>e:\...\sp\src\vscript\squirrel\squirrel\sqobject.cpp(210): warning C4706: assignment within conditional expression
1>e:\...\sp\src\vscript\squirrel\squirrel\sqstate.cpp(244): warning C4100: 'vm' : unreferenced formal parameter
1>e:\...\sp\src\vscript\squirrel\sqstdlib\sqstdblob.cpp(123): warning C4100: 'size' : unreferenced formal parameter
1>e:\...\sp\src\vscript\squirrel\sqstdlib\sqstdio.cpp(118): warning C4100: 'size' : unreferenced formal parameter
1>e:\...\sp\src\vscript\squirrel\sqstdlib\sqstdstring.cpp(389): warning C4100: 'size' : unreferenced formal parameter
1>e:\...\sp\src\vscript\squirrel\sqstdlib\sqstdrex.cpp(567): warning C4611: interaction between '_setjmp' and C++ object destruction is non-portable
1>e:\...\sp\src\vscript\squirrel\sqstdlib\sqstdrex.cpp(382): warning C4706: assignment within conditional expression
1>e:\...\sp\src\vscript\squirrel\sqstdlib\sqstdrex.cpp(407): warning C4706: assignment within conditional expression
1>e:\...\sp\src\vscript\squirrel\sqstdlib\sqstdrex.cpp(444): warning C4706: assignment within conditional expression
1>e:\...\sp\src\vscript\squirrel\sqstdlib\sqstdrex.cpp(452): warning C4706: assignment within conditional expression
1>e:\...\sp\src\vscript\squirrel\sqstdlib\sqstdrex.cpp(479): warning C4706: assignment within conditional expression
1>e:\...\sp\src\vscript\squirrel\sqstdlib\sqstdrex.cpp(486): warning C4706: assignment within conditional expression
1>e:\...\sp\src\vscript\squirrel\sqstdlib\sqstdrex.cpp(550): warning C4702: unreachable code
1>e:\...\sp\src\vscript\squirrel\squirrel\sqtable.h(73): warning C4706: assignment within conditional expression
1>e:\...\sp\src\vscript\squirrel\squirrel\sqvm.cpp(524): warning C4100: 'arg_2' : unreferenced formal parameter
1>e:\...\sp\src\vscript\squirrel\squirrel\sqvm.cpp(1619): warning C4100: 'mm' : unreferenced formal parameter
1>e:\...\sp\src\vscript\squirrel\squirrel\sqvm.cpp(265): warning C4702: unreachable code
1>e:\...\sp\src\vscript\squirrel\squirrel\sqvm.cpp(593): warning C4706: assignment within conditional expression
1>e:\...\sp\src\vscript\squirrel\squirrel\sqvm.cpp(607): warning C4706: assignment within conditional expression
1>e:\...\sp\src\vscript\squirrel\squirrel\sqvm.cpp(1174): warning C4706: assignment within conditional expression
1>e:\...\sp\src\vscript\squirrel\squirrel\sqvm.cpp(1616): warning C4702: unreachable code
```

--- a/sp/src/vscript/vscript.vpc
+++ b/sp/src/vscript/vscript.vpc
@@ -13,9 +13,6 @@ $Configuration
 	$Compiler
 	{
 		$AdditionalIncludeDirectories	"$BASE;.\squirrel\include"
-
-		// Squirrel third party library is full of warnings, so we disable warnings as errors
-		 $TreatWarningsAsErrors	"No"
 	}
 }
 
@@ -29,26 +26,37 @@ $Project "VScript"
 
 		$Folder "squirrel"
 		{
-			$File	".\squirrel\sqstdlib\sqstdaux.cpp"
-			$File	".\squirrel\sqstdlib\sqstdblob.cpp"
-			$File	".\squirrel\sqstdlib\sqstdio.cpp"
-			$File	".\squirrel\sqstdlib\sqstdmath.cpp"
-			$File	".\squirrel\sqstdlib\sqstdrex.cpp"
-			$File	".\squirrel\sqstdlib\sqstdstream.cpp"
-			$File	".\squirrel\sqstdlib\sqstdstring.cpp"
-			$File	".\squirrel\sqstdlib\sqstdsystem.cpp"
-			$File	".\squirrel\squirrel\sqapi.cpp"
-			$File	".\squirrel\squirrel\sqbaselib.cpp"
-			$File	".\squirrel\squirrel\sqclass.cpp"
-			$File	".\squirrel\squirrel\sqcompiler.cpp"
-			$File	".\squirrel\squirrel\sqdebug.cpp"
-			$File	".\squirrel\squirrel\sqfuncstate.cpp"
-			$File	".\squirrel\squirrel\sqlexer.cpp"
-			$File	".\squirrel\squirrel\sqmem.cpp"
-			$File	".\squirrel\squirrel\sqobject.cpp"
-			$File	".\squirrel\squirrel\sqstate.cpp"
-			$File	".\squirrel\squirrel\sqtable.cpp"
-			$File	".\squirrel\squirrel\sqvm.cpp"
+			$File	".\squirrel\sqstdlib\sqstdaux.cpp"		\
+					".\squirrel\sqstdlib\sqstdblob.cpp"		\
+					".\squirrel\sqstdlib\sqstdio.cpp"		\
+					".\squirrel\sqstdlib\sqstdmath.cpp"		\
+					".\squirrel\sqstdlib\sqstdrex.cpp"		\
+					".\squirrel\sqstdlib\sqstdstream.cpp"	\
+					".\squirrel\sqstdlib\sqstdstring.cpp"	\
+					".\squirrel\sqstdlib\sqstdsystem.cpp"	\
+					".\squirrel\squirrel\sqapi.cpp"			\
+					".\squirrel\squirrel\sqbaselib.cpp"		\
+					".\squirrel\squirrel\sqclass.cpp"		\
+					".\squirrel\squirrel\sqcompiler.cpp"	\
+					".\squirrel\squirrel\sqdebug.cpp"		\
+					".\squirrel\squirrel\sqfuncstate.cpp"	\
+					".\squirrel\squirrel\sqlexer.cpp"		\
+					".\squirrel\squirrel\sqmem.cpp"			\
+					".\squirrel\squirrel\sqobject.cpp"		\
+					".\squirrel\squirrel\sqstate.cpp"		\
+					".\squirrel\squirrel\sqtable.cpp"		\
+					".\squirrel\squirrel\sqvm.cpp"
+			{
+				$Configuration
+				{
+					$Compiler
+					{
+						// Squirrel third party library is full of warnings
+						$AdditionalOptions		"$BASE /wd4100 /wd4611 /wd4127 /wd4244 /wd4702 /wd4706 /wd4800"
+						$TreatWarningsAsErrors	"No"
+					}
+				}
+			}
 		}
 	}
 }

--- a/sp/src/vscript/vscript_squirrel.cpp
+++ b/sp/src/vscript/vscript_squirrel.cpp
@@ -802,7 +802,7 @@ bool getVariant(HSQUIRRELVM vm, SQInteger idx, ScriptVariant_t& variant)
 		{
 			return false;
 		}
-		variant = (bool)val;
+		variant = val ? true : false;
 		return true;
 	}
 	case OT_STRING:
@@ -911,7 +911,7 @@ SQInteger function_stub(HSQUIRRELVM vm)
 			SQBool val = 0;
 			if (SQ_FAILED(sq_getbool(vm, i + 2, &val)))
 				return sq_throwerror(vm, "Expected bool");
-			params[i] = (bool)val;
+			params[i] = val ? true : false;
 			break;
 		}
 		case FIELD_CHARACTER:

